client/components/Login/index.jsx

```diff
@@ -72,13 +72,29 @@ class Login extends Component {
             }
         )).json();
 
+        const { permissions } = await (await fetch(
+            '/api/roles/permission'
+        )).json();
+
         if (error) {
             this.setState({ loginError });
             this.props.logOut('');
         } else {
-            Session.create({ username, timeout: Date.now() });
+            this.props.logIn({
+                username,
+                user_id: id,
+                isLoggedIn: true,
+                permissions
+            });
+            Session.create({
+                username,
+                user_id: id,
+                timeout: Date.now(),
+                permissions
+            });
+
             // Step outside of react to force a real reload
-            // after login and session create
+            // force a real request after log-in.
```

Comment:
This makes the comment read "Step outside of react to force a real reload force a real request after log-in."

Reply:
Ah, looks like a messed up rebase. I'll fix it.